h3/src/client.rs

```diff
@@ -288,27 +288,29 @@ where
             .ok_or_else(|| {
                 Code::H3_GENERAL_PROTOCOL_ERROR.with_reason("Did not receive response headers")
             })?;
-
-        let decoded = if let Frame::Headers(ref mut encoded) = frame {
-            qpack::decode_stateless(encoded)?
+        
+        let decoded;
```

Comment:
To make this lint happy, you can write `let decoded = if let Frame::Headers(..)`.

Reply:
I tried this, but then `decoded` has the type of `()` which leads to a type mismatch in this line `let qpack::Decoded { fields, .. } = decoded;`. This is the code:

```rust
        let decoded = if let Frame::Headers(ref mut encoded) = frame {
            match qpack::decode_stateless(encoded, self.inner.max_field_section_size) {
                Err(qpack::DecoderError::HeaderTooLong(cancel_size)) => {
                    self.inner.stop_sending(Code::H3_REQUEST_CANCELLED);
                    return Err(Error::header_too_big(
                        cancel_size,
                        self.inner.max_field_section_size,
                    ));
                }
                Ok(decoded) => decoded,
                Err(e) => return Err(e.into()),
            };
        } else {
            return Err(
                Code::H3_FRAME_UNEXPECTED.with_reason("First response frame is not headers")
            );
        };

        let qpack::Decoded { fields, .. } = decoded;
```
What did i wrong?

Reply:
I think you just need to the remove the semi-colon from the `match` expression.

Reply:
Ah, that helped. Thank you.